### DISCUSSION TITLE
replace_env() -> unable to get value for environment variable "METPLUS_CENSOR_THRESH" error message in GridStat example case

### DISCUSSION THREAD

**Discussion by haiPensKi** [dtcenter/METplus#1593] [Configuration] [answered]
Hi,

I am trying to successfully run the GridStat_fcstGFS_obsCCPA_GRIB example case on NCAR Cheyenne. When I run it, I get an error message that reads: replace_env() -> unable to get value for environment variable "METPLUS_CENSOR_THRESH" . I do not see METPLUS_CENSOR_THRESH  listed in the conf file for this example. Why am I getting this error message and how can I fix it?

conf files:
/glade/u/home/haidenm/GridStat_fcstGFS_obsCCPA_GRIB.conf
/glade/u/home/haidenm/cheyenne.jpresto.conf

Log file:
/glade/u/home/haidenm/logs/metplus.log.20220426153456

**Comment by jprestop** [comment, marked as answer]
Hi @haiPensKi. Thank you for your question.  

Before trying to figure out exactly what is happening here, I see there may be an issue with versions of the software.  At the top of your log file, I see the following:

> Setting [dir] METPLUS_BASE to /glade/p/ral/jntp/MET/METplus/METplus-4.0.0

and 

> Setting [dir] PARM_BASE to /glade/p/ral/jntp/MET/METplus/METplus-4.0.0/parm

Later in the log file, however, I see that a METplus-4.1.1 config file is being used with the MET-10.0.0 version of the MET config file:

> DEBUG 1: Default Config File: /glade/p/ral/jntp/MET/MET_releases/10.0.0/share/met/config/GridStatConfig_default
> DEBUG 1: User Config File: /glade/p/ral/jntp/MET/METplus/METplus-4.1.1/parm/met_config/GridStatConfig_wrapped
> 

If it does not matter which version you run, could you please delete or comment out the following lines in your "/glade/u/home/haidenm/cheyenne.jpresto.conf" config file:

> [dir]
> MET_INSTALL_DIR = /glade/p/ral/jntp/MET/MET_releases/10.0.0

Then, get open a new shell for a clean environment and run the following to load METplus-4.1.1 and MET-10.1.1:

> module use /glade/p/ral/jntp/MET/METplus/modulefiles
> module load metplus/4.1.1

Then, please try rerunning METplus and see if you get the same error.

**Reply by haiPensKi** [reply]
Hi @jprestop,

That worked! Thank you!

**Reply by jprestop** [reply]
That's great news!  Thanks for letting us know.  I'll go ahead and mark my initial response as the "Answer" and lock this conversation to prevent future posts. Our team has decided that once discussions have been answered, and the answer has been confirmed, we'll lock them. We want to encourage users to ask new questions in new discussions rather than posting to old ones. Hopefully that'll make the questions/answers easier for other users to follow. So if/when more issues/questions arise, please feel free to start a new discussion.
